backend/apid/middlewares/authorization_attributes.go

```diff
@@ -93,9 +93,13 @@ func (a AuthorizationAttributes) Then(next http.Handler) http.Handler {
 				attrs.Resource = types.LocalSelfUserResource
 			}
 
-			// Change the resource to LocalSelfUserResource if a user tries to change
-			// its own password
-			if attrs.Verb == "update" && vars["subresource"] == "password" {
+			// check if request comes from webui
+			if strings.Contains(r.URL.Path, "/change_password") {
+				attrs.Resource = types.LocalSelfUserResource
+			}
+
+			switch vars["subresource"] {
+			case "password":
```

Comment:
This code should work in terms of the RBAC system only, and not rely on inspecting URL fragments.

Reply:
@echlebek & @fguimond  have made the change, removed check on URL fragment. it will now check sub resource.